src/pepr/operator/controllers/network/authorizationPolicies.ts

```diff
@@ -0,0 +1,269 @@
+    .toLowerCase()
+    .replace(/\s+/g, "-");
+}
+
+/**
+ * Generates a unique authorization policy name for a given rule.
+ */
+export function generateAuthPolName(rule: Allow | Expose): string {
+  return isExposeRule(rule) ? generateExposeName(rule) : generateAllowName(rule);
+}
+
+/**
+ * Processes an Allow rule to extract its effective source and ports.
+ */
+function processAllowRule(rule: Allow, pkgNamespace: string): { source: Source; ports: string[] } {
+  const ports: string[] = [];
+  if (rule.port !== undefined) ports.push(rule.port.toString());
+  if (rule.ports) ports.push(...rule.ports.map(p => p.toString()));
+
+  let source: Source = {};
+  if (rule.remoteCidr) {
+    source = { ipBlocks: [rule.remoteCidr] };
+  } else if (
+    rule.remoteNamespace === "" ||
+    rule.remoteNamespace === "*" ||
+    rule.remoteGenerated === RemoteGenerated.Anywhere
+  ) {
+    source = {};
+  } else if (rule.remoteGenerated === RemoteGenerated.IntraNamespace) {
+    source = { namespaces: [pkgNamespace] };
+  } else if (rule.remoteNamespace && rule.remoteNamespace !== "" && rule.remoteNamespace !== "*") {
+    source = { namespaces: [rule.remoteNamespace] };
+  } else {
+    source = {};
+  }
+  if (rule.remoteServiceAccount && rule.remoteServiceAccount.trim() !== "") {
+    const nsForSA =
+      rule.remoteNamespace && rule.remoteNamespace.trim() !== ""
+        ? rule.remoteNamespace
+        : pkgNamespace;
+    source = { principals: [`cluster.local/ns/${nsForSA}/sa/${rule.remoteServiceAccount}`] };
+  }
+  return { source, ports };
+}
+
+/**
+ * Processes an Expose rule to extract its effective source and ports.
+ */
+function processExposeRule(rule: Expose): { source: Source; ports: string[] } {
+  const ports: string[] = [];
+  const effectivePort = rule.targetPort ?? rule.port;
+  if (effectivePort !== undefined) {
+    ports.push(effectivePort.toString());
+  }
+  const source =
+    rule.gateway === Gateway.Admin
+      ? { principals: [ADMIN_INGRESS] }
+      : { principals: [TENANT_INGRESS] };
+  return { source, ports };
+}
+
+/**
+ * Helper to determine if an object is empty.
+ */
+function isEmpty(obj: object): boolean {
+  return Object.keys(obj).length === 0;
+}
+
+/**
+ * Helper to build an AuthorizationPolicy from rule details.
+ * If the computed source is empty, the "from" field is omitted.
+ */
+function buildAuthPolicy(
+  pkgName: string,
+  pkgNamespace: string,
+  generation: string,
+  rule: Allow | Expose,
+  source: Source,
+  ports: string[],
+): AuthorizationPolicy {
+  const policyName = `protect-${pkgName}-${generateAuthPolName(rule)}`;
+  const ruleEntry: Rule = {};
+  if (!isEmpty(source)) {
+    ruleEntry.from = [{ source }];
+  }
+  if (ports.length > 0) {
+    ruleEntry.to = [{ operation: { ports } }];
+  }
+  return {
+    apiVersion: "security.istio.io/v1beta1",
+    kind: "AuthorizationPolicy",
+    metadata: {
+      name: policyName,
```

Comment:
I think we'll want to make sure we wrap this in `sanitizeResourceName` (and anywhere else we're making generated names). That just ensures that we truncate and trim anything that would violate k8s resource naming restrictions.